address.go

```diff
@@ -149,6 +176,56 @@ func DecodeAddress(addr string, defaultNet *chaincfg.Params) (Address, error) {
 	}
 }
 
+// ConvertSlpToCashAddress converts an slp formatted address to cash formatted address
+func ConvertSlpToCashAddress(addr Address, params *chaincfg.Params) (Address, error) {
+	var (
+		bchAddr Address
+		err     error
+	)
+	switch a := addr.(type) {
+	case *AddressPubKeyHash:
+		hash := a.Hash160()
+		bchAddr, err = NewAddressPubKeyHash(hash[:], params)
+		if err != nil {
+			return nil, err
+		}
+	case *AddressScriptHash:
+		hash := a.Hash160()
+		bchAddr, err = NewAddressScriptHashFromHash(hash[:], params)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, errors.New("invalid address type")
+	}
+	return bchAddr, nil
+}
+
+// ConvertCashToSlpAddress converts a cash formatted address to slp formatted address
+func ConvertCashToSlpAddress(addr Address, params *chaincfg.Params) (Address, error) {
```

Comment:
@zquestz, in `ConvertSlpToCashAddress` and `ConvertCashToSlpAddress` take Address pointer instead of the current form `ConvertCashToSlpAddress(addr Address, params *chaincfg.Params)`?

Reply:
I guess I'm not sure about whether or not a pointer to an interface is a good idea in golang.  Seems to compile and pass tests, but after quick google search still not sure.

Reply:
ok, so the docs say an interface is just two pointers, one to the the type information and the other to the value passed.  So I guess it doesn't add any value to pass pointer to an interface value.